recipes_source/torch_compile_backend_ipex.rst

```diff
@@ -0,0 +1,157 @@
+
+Usage Example
+~~~~~~~~~~~~~
+
+Train FP32
+----------
+
+.. code:: python
+
+   import torch
+   import torchvision
+
+   LR = 0.001
+   DOWNLOAD = True
+   DATA = 'datasets/cifar10/'
+
+   transform = torchvision.transforms.Compose([
+     torchvision.transforms.Resize((224, 224)),
+     torchvision.transforms.ToTensor(),
+     torchvision.transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+   ])
+   train_dataset = torchvision.datasets.CIFAR10(
+     root=DATA,
+     train=True,
+     transform=transform,
+     download=DOWNLOAD,
+   )
+   train_loader = torch.utils.data.DataLoader(
+     dataset=train_dataset,
+     batch_size=128
+   )
+
+   model = torchvision.models.resnet50()
+   criterion = torch.nn.CrossEntropyLoss()
+   optimizer = torch.optim.SGD(model.parameters(), lr = LR, momentum=0.9)
+   model.train()
+
+   #################### code changes ####################
+   import intel_extension_for_pytorch as ipex
+
+   # Invoke the following API optionally, to apply frontend optimizations
+   model, optimizer = ipex.optimize(model, optimizer=optimizer)
+
+   compile_model = torch.compile(model, backend="ipex")
+   ######################################################
+
+   for batch_idx, (data, target) in enumerate(train_loader):
+       optimizer.zero_grad()
+       output = compile_model(data)
+       loss = criterion(output, target)
+       loss.backward()
+       optimizer.step()
+
+
+Train BF16
+----------
+
+.. code:: python
+
+   import torch
+   import torchvision
+
+   LR = 0.001
+   DOWNLOAD = True
+   DATA = 'datasets/cifar10/'
+
+   transform = torchvision.transforms.Compose([
+     torchvision.transforms.Resize((224, 224)),
+     torchvision.transforms.ToTensor(),
+     torchvision.transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+   ])
+   train_dataset = torchvision.datasets.CIFAR10(
+     root=DATA,
+     train=True,
+     transform=transform,
+     download=DOWNLOAD,
+   )
+   train_loader = torch.utils.data.DataLoader(
+     dataset=train_dataset,
+     batch_size=128
+   )
+
+   model = torchvision.models.resnet50()
+   criterion = torch.nn.CrossEntropyLoss()
+   optimizer = torch.optim.SGD(model.parameters(), lr = LR, momentum=0.9)
+   model.train()
+
+   #################### code changes ####################
+   import intel_extension_for_pytorch as ipex
+
+   # Invoke the following API optionally, to apply frontend optimizations
+   model, optimizer = ipex.optimize(model, dtype=torch.bfloat16, optimizer=optimizer)
+
+   compile_model = torch.compile(model, backend="ipex")
+   ######################################################
+
+   with torch.cpu.amp.autocast():
+       for batch_idx, (data, target) in enumerate(train_loader):
+           optimizer.zero_grad()
+           output = compile_model(data)
+           loss = criterion(output, target)
+           loss.backward()
+           optimizer.step()
+
+
+Inference FP32
+--------------
+
```

Comment:
Same as above - can we add something like: "The following example demonstrates...." and explain what it demonstrates

Reply:
Sure. Added description below.